packages/@sanity/schema/src/sanity/validation/utils/isReactComponentIsh.ts

```diff
@@ -0,0 +1,9 @@
+const REACT_SYM_RE = /^Symbol\(react\..+\)$/
+
+export function isReactComponentIsh(value: any) {
+  const type = typeof value
+  return (
+    type === 'function' ||
+    (typeof value?.$$typeof === 'symbol' && REACT_SYM_RE.test(String(value?.$$typeof)))
+  )
+}
```

Comment:
Could we use [react-is](https://www.npmjs.com/package/react-is) instead?

Reply:
ah, realized now. It allows too much (e.g. strings), which doesn't make all that much sense for an input-component.

Reply:
(added a code comment about this)